frontends/election-manager/src/utils/write_ins.ts

```diff
@@ -34,3 +50,142 @@ export function getWriteInCountsByContestAndCandidate(
     )
   );
 }
+
+export function filterWriteInCountsByParty(
+  counts: CountsByContestAndCandidateName,
+  election: Election,
+  partyId?: PartyId
+): CountsByContestAndCandidateName {
+  if (!partyId) return counts;
+
+  const filteredCounts = new Map<ContestId, Map<string, number>>();
+  for (const [contestId, contestCounts] of counts) {
+    const contest = election.contests.find((c) => c.id === contestId);
+    if (contest && contest.partyId === partyId) {
+      filteredCounts.set(contestId, contestCounts);
+    }
+  }
+  return filteredCounts;
+}
+
+// ExternalTally write-in candidate id creation
+export function getAdjudicatedWriteInCandidateId(
+  name: string,
+  manual: boolean
+): string {
+  return `write-in-(${name})${manual ? '-manual' : ''}`;
```

Comment:
oh I see this is where the ID comes from

Reply:
Does that ID format seem OK?